File: frontend/src/api/modules/operation.ts
/**
 * 运维管理 API
 */
import request from '@/utils/request'
import type { ResponseModel, PageParams } from './types'

// ==================== 类型定义 ====================

/** 工单状态 */
export type WorkOrderStatus = 'pending' | 'assigned' | 'processing' | 'completed' | 'closed' | 'cancelled'

/** 工单类型 */
export type WorkOrderType = 'fault' | 'maintenance' | 'inspection' | 'change' | 'other'

/** 工单优先级 */
export type WorkOrderPriority = 'critical' | 'high' | 'medium' | 'low'

/** 巡检状态 */
export type InspectionStatus = 'pending' | 'in_progress' | 'completed' | 'overdue'

/** 工单信息 */
export interface WorkOrder {
  id: number
  order_no: string
  title: string
  description?: string
  order_type: WorkOrderType
  priority: WorkOrderPriority
  status: WorkOrderStatus
  device_id?: number
  device_name?: string
  location?: string
  reporter?: string
  reporter_phone?: string
  assignee?: string
  deadline?: string
  created_at: string
  assigned_at?: string
  started_at?: string
  completed_at?: string
  solution?: string
  root_cause?: string
}

/** 工单创建参数 */
export interface WorkOrderCreate {
  title: string
  description?: string
  order_type: WorkOrderType
  priority: WorkOrderPriority
  device_id?: number
  location?: string
  reporter?: string
  reporter_phone?: string
  deadline?: string
}

/** 工单更新参数 */
export interface WorkOrderUpdate {
  title?: string
  description?: string
  order_type?: WorkOrderType
  priority?: WorkOrderPriority
  device_id?: number
  location?: string
  reporter?: string
  reporter_phone?: string
  deadline?: string
}

/** 工单日志 */
export interface WorkOrderLog {
  id: number
  work_order_id: number
  action: string
  content: string
  operator?: string
  created_at: string
}

/** 巡检计划 */
export interface InspectionPlan {
  id: number
  plan_name: string
  description?: string
  plan_type: string
  frequency: string
  start_date: string
  end_date?: string
  devices?: number[]
  locations?: string[]
  inspector?: string
  checklist?: string[]
  is_active: boolean
  created_at: string
  updated_at: string
}

/** 巡检计划创建参数 */
export interface InspectionPlanCreate {
  plan_name: string
  description?: string
  plan_type: string
  frequency: string
  start_date: string
  end_date?: string
  devices?: number[]
  locations?: string[]
  inspector?: string
  checklist?: string[]
  is_active?: boolean
}

/** 巡检任务 */
export interface InspectionTask {
  id: number
  plan_id: number
  plan_name?: string
  task_no: string
  status: InspectionStatus
  inspector?: string
  scheduled_time: string
  started_at?: string
  completed_at?: string
  devices?: number[]
  locations?: string[]
  checklist?: string[]
  results?: Record<string, any>
  remarks?: string
  created_at: string
}

/** 巡检任务创建参数 */
export interface InspectionTaskCreate {
  plan_id: number
  inspector?: string
  scheduled_time: string
  devices?: number[]
  locations?: string[]
  checklist?: string[]
}

/** 知识库 */
export interface Knowledge {
  id: number
  title: string
  category: string
  content: string
  keywords?: string[]
  author?: string
  views: number
  likes: number
  is_published: boolean
  created_at: string
  updated_at: string
}

/** 知识库创建参数 */
export interface KnowledgeCreate {
  title: string
  category: string
  content: string
  keywords?: string[]
  author?: string
  is_published?: boolean
}

/** 运维统计 */
export interface OperationStatistics {
  total_workorders: number
  pending_workorders: number
  processing_workorders: number
  completed_workorders: number
  overdue_workorders: number
  total_inspections: number
  pending_inspections: number
  completed_inspections: number
  overdue_inspections: number
  total_knowledge: number
  workorder_completion_rate: number
  inspection_completion_rate: number
  average_resolution_time: number
}

// ==================== 工单管理 API ====================

/** 获取工单列表 */
export function getWorkOrders(params?: PageParams & {
  status?: WorkOrderStatus
  priority?: WorkOrderPriority
  order_type?: WorkOrderType
  keyword?: string
}) {
  return request.get<ResponseModel<WorkOrder[]>>('/v1/operation/workorders', { params })
}

/** 获取工单详情 */
export function getWorkOrder(id: number) {
  return request.get<ResponseModel<WorkOrder>>(`/v1/operation/workorders/${id}`)
}

/** 创建工单 */
export function createWorkOrder(data: WorkOrderCreate) {
  return request.post<ResponseModel<WorkOrder>>('/v1/operation/workorders', data)
}

/** 更新工单 */
export function updateWorkOrder(id: number, data: WorkOrderUpdate) {
  return request.put<ResponseModel<WorkOrder>>(`/v1/operation/workorders/${id}`, data)
}

/** 删除工单 */
export function deleteWorkOrder(id: number) {
  return request.delete<ResponseModel>(`/v1/operation/workorders/${id}`)
}

/** 分配工单 */
export function assignWorkOrder(id: number, assignee: string) {
  return request.post<ResponseModel<WorkOrder>>(`/v1/operation/workorders/${id}/assign`, { assignee })
}

/** 开始处理工单 */
export function startWorkOrder(id: number) {
  return request.post<ResponseModel<WorkOrder>>(`/v1/operation/workorders/${id}/start`)
}

/** 完成工单 */
export function completeWorkOrder(id: number, solution: string, root_cause?: string) {
  return request.post<ResponseModel<WorkOrder>>(`/v1/operation/workorders/${id}/complete`, { solution, root_cause })
}

/** 获取工单日志 */
export function getWorkOrderLogs(id: number) {
  return request.get<ResponseModel<WorkOrderLog[]>>(`/v1/operation/workorders/${id}/logs`)
}

/** 添加工单日志 */
export function addWorkOrderLog(id: number, action: string, content: string, operator?: string) {
  return request.post<ResponseModel<WorkOrderLog>>(`/v1/operation/workorders/${id}/logs`, { action, content, operator })
}

// ==================== 巡检计划 API ====================

/** 获取巡检计划列表 */
export function getInspectionPlans(params?: PageParams & {
  plan_type?: string
  is_active?: boolean
  keyword?: string
}) {
  return request.get<ResponseModel<InspectionPlan[]>>('/v1/operation/inspection-plans', { params })
}

/** 获取巡检计划详情 */
export function getInspectionPlan(id: number) {
  return request.get<ResponseModel<InspectionPlan>>(`/v1/operation/inspection-plans/${id}`)
}

/** 创建巡检计划 */
export function createInspectionPlan(data: InspectionPlanCreate) {
  return request.post<ResponseModel<InspectionPlan>>('/v1/operation/inspection-plans', data)
}

/** 更新巡检计划 */
export function updateInspectionPlan(id: number, data: Partial<InspectionPlanCreate>) {
  return request.put<ResponseModel<InspectionPlan>>(`/v1/operation/inspection-plans/${id}`, data)
}

/** 删除巡检计划 */
export function deleteInspectionPlan(id: number) {
  return request.delete<ResponseModel>(`/v1/operation/inspection-plans/${id}`)
}

// ==================== 巡检任务 API ====================

/** 获取巡检任务列表 */
export function getInspectionTasks(params?: PageParams & {
  plan_id?: number
  status?: InspectionStatus
  inspector?: string
  keyword?: string
}) {
  return request.get<ResponseModel<InspectionTask[]>>('/v1/operation/inspection-tasks', { params })
}

/** 获取巡检任务详情 */
export function getInspectionTask(id: number) {
  return request.get<ResponseModel<InspectionTask>>(`/v1/operation/inspection-tasks/${id}`)
}

/** 创建巡检任务 */
export function createInspectionTask(data: InspectionTaskCreate) {
  return request.post<ResponseModel<InspectionTask>>('/v1/operation/inspection-tasks', data)
}

/** 更新巡检任务 */
export function updateInspectionTask(id: number, data: Partial<InspectionTaskCreate>) {
  return request.put<ResponseModel<InspectionTask>>(`/v1/operation/inspection-tasks/${id}`, data)
}

/** 删除巡检任务 */
export function deleteInspectionTask(id: number) {
  return request.delete<ResponseModel>(`/v1/operation/inspection-tasks/${id}`)
}

/** 开始巡检任务 */
export function startInspectionTask(id: number) {
  return request.post<ResponseModel<InspectionTask>>(`/v1/operation/inspection-tasks/${id}/start`)
}

/** 完成巡检任务 */
export function completeInspectionTask(id: number, results: Record<string, any>, remarks?: string) {
  return request.post<ResponseModel<InspectionTask>>(`/v1/operation/inspection-tasks/${id}/complete`, { results, remarks })
}

// ==================== 知识库 API ====================

/** 获取知识库列表 */
export function getKnowledgeList(params?: PageParams & {
  category?: string
  is_published?: boolean
  keyword?: string
}) {
  return request.get<ResponseModel<Knowledge[]>>('/v1/operation/knowledge', { params })
}

/** 获取知识库详情 */
export function getKnowledge(id: number) {
  return request.get<ResponseModel<Knowledge>>(`/v1/operation/knowledge/${id}`)
}

/** 创建知识库 */
export function createKnowledge(data: KnowledgeCreate) {
  return request.post<ResponseModel<Knowledge>>('/v1/operation/knowledge', data)
}

/** 更新知识库 */
export function updateKnowledge(id: number, data: Partial<KnowledgeCreate>) {
  return request.put<ResponseModel<Knowledge>>(`/v1/operation/knowledge/${id}`, data)
}

/** 删除知识库 */
export function deleteKnowledge(id: number) {
  return request.delete<ResponseModel>(`/v1/operation/knowledge/${id}`)
}

// ==================== 统计 API ====================

/** 获取运维统计 */
export function getOperationStatistics() {
  return request.get<ResponseModel<OperationStatistics>>('/v1/operation/statistics')
}

File: backend/app/models/operation.py
"""
运维管理模型
"""
from datetime import datetime
from enum import Enum as PyEnum
from sqlalchemy import Column, Integer, String, Boolean, Text, DateTime, ForeignKey, Enum
from sqlalchemy.orm import relationship

from ..core.database import Base


# ==================== 枚举定义 ====================

class WorkOrderStatus(str, PyEnum):
    """工单状态枚举"""
    pending = "待处理"
    assigned = "已派单"
    processing = "处理中"
    completed = "已完成"
    closed = "已关闭"
    cancelled = "已取消"


class WorkOrderType(str, PyEnum):
    """工单类型枚举"""
    fault = "故障报修"
    maintenance = "日常维护"
    inspection = "巡检任务"
    change = "变更请求"
    other = "其他"


class WorkOrderPriority(str, PyEnum):
    """工单优先级枚举"""
    critical = "紧急"
    high = "高"
    medium = "中"
    low = "低"


class InspectionStatus(str, PyEnum):
    """巡检状态枚举"""
    pending = "待巡检"
    in_progress = "巡检中"
    completed = "已完成"
    overdue = "已逾期"


# ==================== 工单模型 ====================

class WorkOrder(Base):
    """工单表"""
    __tablename__ = "work_orders"

    id = Column(Integer, primary_key=True, autoincrement=True)
    order_no = Column(String(50), unique=True, nullable=False, comment="工单编号")
    title = Column(String(200), nullable=False, comment="工单标题")
    description = Column(Text, comment="工单描述")
    order_type = Column(Enum(WorkOrderType), default=WorkOrderType.other, comment="工单类型")
    priority = Column(Enum(WorkOrderPriority), default=WorkOrderPriority.medium, comment="优先级")
    status = Column(Enum(WorkOrderStatus), default=WorkOrderStatus.pending, comment="工单状态")
    device_id = Column(Integer, comment="关联设备ID")
    device_name = Column(String(100), comment="设备名称")
    location = Column(String(200), comment="位置")
    reporter = Column(String(100), comment="报修人")
    reporter_phone = Column(String(50), comment="报修人电话")
    assignee = Column(String(100), comment="处理人")
    created_at = Column(DateTime, default=datetime.now, comment="创建时间")
    assigned_at = Column(DateTime, comment="派单时间")
    started_at = Column(DateTime, comment="开始处理时间")
    completed_at = Column(DateTime, comment="完成时间")
    closed_at = Column(DateTime, comment="关闭时间")
    deadline = Column(DateTime, comment="截止时间")
    solution = Column(Text, comment="解决方案")
    root_cause = Column(Text, comment="根本原因")
    remarks = Column(Text, comment="备注")
    satisfaction = Column(Integer, comment="满意度(1-5)")
    feedback = Column(Text, comment="反馈")

    # 关系
    logs = relationship("WorkOrderLog", back_populates="work_order", cascade="all, delete-orphan")


class WorkOrderLog(Base):
    """工单日志表"""
    __tablename__ = "work_order_logs"

    id = Column(Integer, primary_key=True, autoincrement=True)
    order_id = Column(Integer, ForeignKey("work_orders.id"), nullable=False, comment="工单ID")
    action = Column(String(50), comment="操作类型")
    content = Column(Text, comment="操作内容")
    operator = Column(String(100), comment="操作人")
    created_at = Column(DateTime, default=datetime.now, comment="创建时间")

    # 关系
    work_order = relationship("WorkOrder", back_populates="logs")


# ==================== 巡检计划模型 ====================

class InspectionPlan(Base):
    """巡检计划表"""
    __tablename__ = "inspection_plans"

    id = Column(Integer, primary_key=True, autoincrement=True)
    name = Column(String(100), nullable=False, comment="计划名称")
    description = Column(Text, comment="计划描述")
    frequency = Column(String(50), comment="巡检频率(daily/weekly/monthly)")
    location = Column(String(200), comment="巡检位置")
    check_items = Column(Text, comment="检查项目(JSON字符串)")
    assignee = Column(String(100), comment="负责人")
    is_active = Column(Boolean, default=True, comment="是否启用")
    created_at = Column(DateTime, default=datetime.now, comment="创建时间")
    updated_at = Column(DateTime, default=datetime.now, onupdate=datetime.now, comment="更新时间")

    # 关系
    tasks = relationship("InspectionTask", back_populates="plan", cascade="all, delete-orphan")


class InspectionTask(Base):
    """巡检任务表"""
    __tablename__ = "inspection_tasks"

    id = Column(Integer, primary_key=True, autoincrement=True)
    plan_id = Column(Integer, ForeignKey("inspection_plans.id"), comment="巡检计划ID")
    task_no = Column(String(50), unique=True, nullable=False, comment="任务编号")
    status = Column(Enum(InspectionStatus), default=InspectionStatus.pending, comment="任务状态")
    assignee = Column(String(100), comment="执行人")
    scheduled_date = Column(DateTime, comment="计划执行日期")
    started_at = Column(DateTime, comment="开始时间")
    completed_at = Column(DateTime, comment="完成时间")
    result = Column(Text, comment="巡检结果(JSON)")
    abnormal_count = Column(Integer, default=0, comment="异常数量")
    remarks = Column(Text, comment="备注")
    created_at = Column(DateTime, default=datetime.now, comment="创建时间")

    # 关系
    plan = relationship("InspectionPlan", back_populates="tasks")


# ==================== 知识库模型 ====================

class KnowledgeBase(Base):
    """知识库表"""
    __tablename__ = "knowledge_base"

    id = Column(Integer, primary_key=True, autoincrement=True)
    title = Column(String(200), nullable=False, comment="标题")
    category = Column(String(100), comment="分类")
    content = Column(Text, comment="内容")
    tags = Column(String(500), comment="标签")
    view_count = Column(Integer, default=0, comment="查看次数")
    is_published = Column(Boolean, default=False, comment="是否发布")
    author = Column(String(100), comment="作者")
    created_at = Column(DateTime, default=datetime.now, comment="创建时间")
    updated_at = Column(DateTime, default=datetime.now, onupdate=datetime.now, comment="更新时间")
